src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -25,7 +26,10 @@ namespace StreamJsonRpc
     /// <summary>
     /// Uses Newtonsoft.Json serialization to serialize <see cref="JsonRpcMessage"/> as JSON (text).
     /// </summary>
-    public class JsonMessageFormatter : IJsonRpcAsyncMessageTextFormatter
+    /// <remarks>
+    /// Each instance of this class may only be used with a single <see cref="JsonRpc" /> instance.
+    /// </remarks>
+    public class JsonMessageFormatter : MessageFormatterHelper, IJsonRpcAsyncMessageTextFormatter, IJsonRpcInstanceContainer
```

Comment:
>MessageFormatterHelper, [](start = 40, length = 23)

Please refactor this so that it's a helper class rather than a base class. Base classes have high impact and as formatters can only derive from _one_ base class, a class that assists in adding functionality such as $/progress should not take the premium one spot as base class. As a helper class, it becomes an instance field on `JsonMessageFormatter` and thus scales better when we add other such helper classes.
Hint: we very likely will add other helpers. For example: we want to start marshaling System.IO.Stream in messages, sort of like how we're marshaling IProgress<T>.